src/formula_gen.cpp

```diff
@@ -392,14 +408,14 @@ std::pair<bool, Formula> FormulaGenerator::generate(const Program& p,
     }
   }
   if (keys.size() > 2) {
-    return result;
+    return false;
   }
   if (recursive.size() > 1) {
-    return result;
+    return false;
   }
   for (auto r : recursive) {
     if (deps.count(r) > 1) {
```

Comment:
Maybe greater than zero?

Reply:
It's actually about recursive formulas, not programs. We support some basic recursive formulas already, but not multiple functions calling each other recursively. This is what this check is about.